features/netsocket/UDPSocket.h

```diff
@@ -56,6 +56,8 @@ class UDPSocket : public Socket {
      */
     virtual ~UDPSocket();
 
+    int join_multicast_group(SocketAddress* address);
```

Comment:
Needs documentation?

Also SocketAddress is a first class object, so it should be passed around by value or reference. The should be
``` cpp
int join_multicast_group(const SocketAddress &address);
``` 

That way you can pass implicitly cast strings
``` cpp
TCPSocket s
s.join_multicast_group("250.0.0.10");
```